layer/nrlf/nrlf/core/repository.py

```diff
@@ -525,7 +525,15 @@ def _delete(id):
         if len(delete_pks) >= MAX_TRANSACT_ITEMS:
             raise TooManyItemsError("Too many items to process in one transaction")
         transact_items = [_delete(id) for id in delete_pks] + [_put()]
-        return self.dynamodb.transact_write_items(TransactItems=transact_items)
+        try:
+            return self.dynamodb.transact_write_items(TransactItems=transact_items)
+        except ClientError as error:
+            if "CancellationReasons" in error.response:
+                reasons = error.response["CancellationReasons"]
+                # Last reason relates to the single PUT and indicates duplicate item
+                if reasons[-1]["Code"] == "ConditionalCheckFailed":
+                    raise DuplicateError("Condition check failed - Duplicate item")
+            raise error
```

Comment:
because the `reasons[-1]` and `[put()]` are implicitly coupled this will be prone to a bug in future. I can imagine if, for some reason, someone change L527 to:

```python
transact_items = [_delete(id) for id in delete_pks] + [_put()] + [some_other_transaction()]
```

then `reasons[-1]` wouldn't be related to the put error. This is also generally the problem with `handle_dynamodb_errors`. Since the transactions are annotated with `Put` and `Delete`, I might suggest in this case that we pull the index out of `transact_item` e.g.

```python
def _is_put(transaction: dict):
    action, = transaction.keys()
    return action == "PUT"

put_idx = next(idx for idx, item in enumerate(transact_items) if _is_put(transaction=item))
```

and then 

```python
reasons[put_idx]
```

Reply:
Or we add a comment above `[put()]`